graph/graph.go

```diff
@@ -141,14 +142,92 @@ func (graph *Graph) restore() error {
 		}
 	}
 
-	baseIds, err := graph.restoreBaseImages()
+	graph.idIndex = truncindex.NewTruncIndex(ids)
+	logrus.Debugf("Restored %d elements", len(ids))
+	return nil
+}
+
+func (graph *Graph) RestoreCustomImages(ts *TagStore) error {
```

Comment:
Also note that `RestoreCustomImages` might list the created image ids so they can be indexed or protected by the caller:

```
type ImageRestorer {
    RestoreCustomImages(tagger Tagger) ([]string, error)
}
```